src/platform/silabs/SiWx917/wifi/ethernetif.cpp

```diff
@@ -268,17 +283,23 @@ static err_t low_level_output(struct netif * netif, struct pbuf * p)
  * @return
  *    None
  ******************************************************************************/
-void wfx_host_received_sta_frame_cb(uint8_t * buf, int len)
+sl_status_t sl_si91x_host_process_data_frame(sl_wifi_interface_t interface, sl_wifi_buffer_t * buffer)
 {
+    struct pbuf * pbuf_packet;
+    void * packet;
     struct netif * ifp;
+    sl_si91x_packet_t * rsi_pkt;
+    packet  = sl_si91x_host_get_buffer_data(buffer, 0, NULL);
+    rsi_pkt = (sl_si91x_packet_t *) packet;
 
     /* get the network interface for STATION interface,
      * and forward the received frame buffer to LWIP
      */
     if ((ifp = wfx_get_netif(SL_WFX_STA_INTERFACE)) != (struct netif *) 0)
     {
-        low_level_input(ifp, buf, len);
+        low_level_input(ifp, rsi_pkt->data, rsi_pkt->length);
     }
+    return SL_STATUS_OK;
```

Comment:
This function cannot fail ?

Reply:
Yes this is required because SDK code expects a return after successful completion